src/open_inwoner/accounts/views/login.py

```diff
@@ -0,0 +1,369 @@
+from collections import OrderedDict
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth import REDIRECT_FIELD_NAME, get_user_model, login as auth_login
+from django.contrib.auth.views import LoginView
+from django.core.signing import BadSignature, SignatureExpired, TimestampSigner
+from django.http import HttpResponse, HttpResponseRedirect
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.utils.http import is_safe_url, urlencode
+from django.utils.translation import gettext as _
+from django.views.decorators.cache import never_cache
+from django.views.generic import FormView, View
+
+from formtools.wizard.views import SessionWizardView
+from oath import totp
+
+from open_inwoner.configurations.models import SiteConfiguration
+from open_inwoner.utils.mixins import ThrottleMixin
+from open_inwoner.utils.views import LogMixin
+
+from ..forms import PhoneNumberForm, VerifyTokenForm
+from ..gateways import GatewayError, gateway
+
+signer = TimestampSigner()
+
+ADD_PHONE_NUMBER_TEMPLATES = {
+    "phonenumber": "registration/add_phone_number_1.html",
+    "verify": "registration/add_phone_number_2.html",
+}
+
+
+class CustomLoginView(LogMixin, LoginView):
+    def form_valid(self, form):
+        config = SiteConfiguration.get_solo()
+
+        if not config.login_2fa_sms:
+            return super().form_valid(form)
+
+        user = form.get_user()
+
+        redirect_to = self.request.POST.get(
+            self.redirect_field_name, self.request.GET.get(self.redirect_field_name, "")
+        )
+
+        params = {
+            self.redirect_field_name: redirect_to,
+            "user": signer.sign(user.pk),
+        }
+
+        if not user.phonenumber:
+            # Redirect to add mobile phone number view
+            return HttpResponseRedirect(
+                reverse("add_phone_number") + "?" + urlencode(params)
+            )
```

Comment:
In Django there is also the `redirect()` shortcut that is slightly nicer then HttpResponseRedirect, and usually we like `furl()` but in this case it's probably fine as is.